### PR TITLE
network: tc: add HTB support

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
cc @ssahani.

File: src/network/tc/tc.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        _TC_KIND_INVALID = -1,
+} TrafficControlKind;
+
+typedef struct TrafficControl {
+        TrafficControlKind kind;
+} TrafficControl;
+
+/* For casting a tc into the various tc kinds */
+#define DEFINE_TC_CAST(UPPERCASE, MixedCase)                           \
+        static inline MixedCase* TC_TO_##UPPERCASE(TrafficControl *tc) {                    \
```
Comment (ssahani) [inline src/network/tc/tc.h:20]: indention

Comment (yuwata) [reply src/network/tc/tc.h:20]: Hmm, we use such indentation rule at many places, e.g., netdev.h and string-table.h.

**Comment by ssahani** [comment]
Thanks for working on it. Looks good not tested . We can wait for the CI and rebase.

**Pull request closed without merging**